drivers/spi/spi_nrfx_spi.c

```diff
@@ -271,23 +272,49 @@ static int init_spi(struct device *dev, const nrfx_spi_config_t *config)
 	return 0;
 }
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int spi_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
+				void *context, device_pm_cb cb, void *arg)
+{
+	static u32_t current_state = DEVICE_PM_ACTIVE_STATE;
+
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		u32_t new_state = *((const u32_t *)context);
+
+		if (new_state != current_state) {
+			switch (new_state) {
+			case DEVICE_PM_ACTIVE_STATE:
+				init_spi(dev);
+				configure(dev, get_dev_data(dev)->ctx.config);
```

Comment:
This is not safe. Before `tranceive` (and consequently `configure`) is called for the first time, `ctx.config` is `NULL`. And nothing prevents `spi_nrfx_pm_control` from being called at that time. You should change this line to:
```
get_dev_data(dev)->ctx.config = NULL;
```
preferably with a comment explaining that it is done to force reconfiguration of the peripheral before the next transfer.